src/pages/ModuleEditor/ModuleEditor.tsx

```diff
@@ -87,6 +92,12 @@ export default function ModuleEditor({match: {params: {moduleId}}}: Props) {
     LoadModule();
   }, [moduleId]);
 
+  React.useEffect(() => {
+    if (userRole === 'Admin' || (userRole === 'Creator' && userId === initialValues.ownerId)) {
+      setCanDisable(true);
+    }
+  }, [initialValues]);
+
```

Comment:
No need to use state here, you can calculate this on every render, it's cheap to do so.
Write it as a function so the current values can be used, not the initial values, the initial values are always going to have a user id of 0.

```suggestion
  const canDisable = userRole === 'Admin' || (userRole === 'Creator' &&  user?.id === initialValues.ownerId);

```